### PR TITLE
release-tools: brush up the documentation of the release process

### PR DESCRIPTION
**Pull request by mspncp**
cc @mattcaswell: A byproduct of my preparation for tuesday. ;-)

Note that this pr not only improves layout and documentation, but also the  workflow itself (e.g., by introducing git worktrees and setting the PATH variable.)

_(You can use the `grip` (GitHub Readme Instant Preview) viewer to preview the files locally.)_

### CODE REVIEW

**Comment by mspncp** [comment]
> _(You can use the `grip` (GitHub Readme Instant Preview) viewer to preview the files locally.)_

Or view them online here:

* [README.md](https://github.com/openssl/tools/blob/7f1121be80e5273092c605ecba1eea1b86617222/release-tools/README.md)
* [MKRELEASE.md](https://github.com/openssl/tools/blob/7f1121be80e5273092c605ecba1eea1b86617222/release-tools/MKRELEASE.md)

File: release-tools/MKRELEASE.md [hunk 1]
```diff
@@ -3,76 +3,87 @@
...
-- OPENSSL_TAR<br>
-  The tar command to use when creating the tarball. Default is `tar`.
+        OPENSSL                    [openssl]
 
-- OPENSSL_GPG<br>
-  The gpg command to use when signing a tarball or announcement.
-  The default is `gpg` which will use gpg with the default key.
-  If you wish to use a different, key set OPENSSL_GPG to include
-  appropriate options.
+Path to openssl utility to use. Default is
```
Comment (romen) [inline release-tools/MKRELEASE.md:52]: "Default is"... missing! :smile:

File: release-tools/MKRELEASE.md [hunk 2]
```diff
@@ -3,76 +3,87 @@
...
+        OPENSSL_GPG                [gpg]
 
-- OPENSSL_SCP_HOST<br>
-  The host (and optional username) needed to upload files.
-  The default is `dev.openssl.org`, but you might want to change this to
-  `username@dev.openssl.org`
+The gpg command to use when signing a tarball or announcement.
+The default is `gpg` which will use gpg with the default key.
+If you wish to use a different, key set OPENSSL_GPG to include
+appropriate options.
```
Comment (romen) [inline release-tools/MKRELEASE.md:63]: sidenote: Maybe include an example?

File: release-tools/MKRELEASE.md (hunk 1 above)
Comment (romen) [reply release-tools/MKRELEASE.md:52]: Also, should this start with "The path" as done for most of the items in this list?

File: release-tools/MKRELEASE.md [hunk 3]
```diff
@@ -3,76 +3,87 @@
...
 
-- OPENSSL_SCP_DIR<br>
-  The directory to upload files to.
-  Normally this wont be changed from the default which is
-  `$OPENSSL_SCP_HOST:~openssl/dist/new`
-  This is a holding area on dev.openssl.org where distributions are uploaded
-  temporarily before being moved to the web and ftp directories.
+        OPENSSL_GPG_TAR            [$OPENSSL_GPG --use-agent -sba]
+
+Command to use to sign a tarball.
```
Comment (romen) [inline release-tools/MKRELEASE.md:67]: side note: "The command" to be coherent with the rest of the list?

File: release-tools/MKRELEASE.md [hunk 4]
```diff
@@ -3,76 +3,87 @@
...
+
+        OPENSSL_GPG_ANNOUNCE       [$OPENSSL_GPG --use-agent -sta --clearsign]
+
+The command to use to sign a tarball.
```
Comment (romen) [inline release-tools/MKRELEASE.md:71]: What is the difference between `OPENSSL_GPG_TAR` and `OPENSSL_GPG_ANNOUNCE`, are they both used for signing a tarball?

File: release-tools/MKRELEASE.md [hunk 5]
```diff
@@ -3,76 +3,87 @@
...
+
+        OPENSSL_SCP                [scp]
```
Comment (romen) [inline release-tools/MKRELEASE.md:73]: `scp` uses `SSH-1` and there are other concerns as well about using it securely (e.g. CVE-2019-6111).

The official [OpenSSH release notes](https://www.openssh.com/txt/release-8.0) say

```
The scp protocol is outdated, inflexible and not readily fixed. We
recommend the use of more modern protocols like sftp and rsync for
file transfer instead.
```

We should probably change the default to `rsync` and make sure the script can work with it.

Ok, probably this is a matter for another separate PR.

**Comment by levitte** [comment]
One small note.  [gitglossary(7)](https://linux.die.net/man/7/gitglossary) says "working tree", not "worktree".  I would normally not care, but I've recently had an instance with someone not understanding "worktree", so...

**Comment by mspncp** [comment]
> One small note. [gitglossary(7)](https://linux.die.net/man/7/gitglossary) says "working tree", not "worktree". I would normally not care, but I've recently had an instance with someone not understanding "worktree", so...

@levitte good point. I was simply using the subcommand name, but even the [git-worktree](https://git-scm.com/docs/git-worktree) manual page talks about working trees.

@romen I see you also added some comments. Sorry for neglecting them, this PR went completely off my radar...

**Comment by levitte** [comment]
For some reason, I failed to notice that you worked on `README.md` too.  There's a competing PR, #75, that moves those instructions to a more prominent location, takes into account the new release script, yada yada.  That gets us two places to discuss changes, maybe we should decide on one to avoid double work?

**Comment by mspncp** [comment]
Yeah, I noticed that, too, and wondered what to do about it. Since my PR has been lying around for a while an yours is active, I don't mind if you simply skim over mine and 'steal' what's worth taking. In that case, I'll close my PR in favour of yours.

**Comment by levitte** [comment]
> In that case, I'll close my PR in favour of yours.

Er, I have *zero* plans to touch MKRELEASE.md any further.  I view that as separate from the README

**Comment by levitte** [comment]
> I don't mind if you simply skim over mine and 'steal' what's worth taking.

Thanks for the offer, I'll do that for the README stuff

**Comment by mspncp** [comment]
Ok, I have to admit that I didn't have the time yet to take a closer look at your PR. Just go one with yours, I'll wait and see what's left of mine.

**Comment by levitte** [comment]
> Just go one with yours

Done, and it's open for discussion if you feel inclined.  Mostly, I transferred the idea of using `git worktree`

File: release-tools/MKRELEASE.md [hunk 6]
```diff
@@ -3,76 +3,87 @@
...
+`--no-clean`, the release process will run `git clean -x -d -f`, which will
+delete ALL untracked files from the worktree.
+See the [README](README.md#prepare-the-worktrees) file for a suggestion how to
+prepare the worktrees.
 
-From the branch directory if you run:
+Change into the worktree and run:
 
-        perl /path/to/mkrelease.pl --reviewer=name
+        mkrelease.pl --reviewer=name
```
Comment (levitte) [inline release-tools/MKRELEASE.md:39]: Elsewhere, we're sticking to 4 space indent for verbatim stuff...  How about here (and I do mean throughout the file)?

**Pull request closed without merging**